zug/transducer/transducer.hpp

```diff
@@ -209,4 +209,26 @@ class transducer
     xform_t xform_;
 };
 
+namespace detail {
+
+template <typename T>
+constexpr bool is_transducer_v = false;
+
+template <typename InputT, typename OutputT>
+constexpr bool is_transducer_v<transducer<InputT, OutputT>> = true;
+
+} // namespace detail
+
+template <
+    typename Lhs,
+    typename Rhs,
+    std::enable_if_t<!detail::is_composed_v<std::decay_t<Lhs>> &&
+                     !detail::is_composed_v<std::decay_t<Rhs>> &&
+                     (detail::is_transducer_v<std::decay_t<Lhs>> ||
+                      detail::is_transducer_v<std::decay_t<Rhs>>)>* = nullptr>
```

Comment:
I do not like having to deal with these rules here. I was thinking of adding a mixin to make a function composable. So basically the transducer definition would look like:
```
template <...>
struct transducer : composable_mixin<transducer<...>>
```

Reply:
You got it!

Reply:
Nice!